drivers/display/ssd1306.c

```diff
@@ -0,0 +1,478 @@
+		  const void *buf)
+{
+	if (desc->pitch < desc->width) {
+		LOG_ERR("Pitch is smaller then width");
+		return -1;
+	}
+
+	if (buf == NULL || desc->buf_size == 0) {
+		LOG_ERR("Display buffer is not available");
+		return -1;
+	}
+
+	if (desc->pitch > desc->width) {
+		LOG_ERR("Unsupported mode");
+		return -1;
+	}
+
+	if (x != 0 && y != 0) {
+		LOG_ERR("Unsupported origin");
+		return -1;
+	}
+
+#if defined(CONFIG_SSD1306_DEFAULT)
+	struct ssd1306_data *driver = dev->driver_data;
+
+	u8_t cmd_buf[] = {
+		SSD1306_CONTROL_BYTE_CMD,
+		SSD1306_SET_MEM_ADDRESSING_MODE,
+		SSD1306_CONTROL_BYTE_CMD,
+		SSD1306_ADDRESSING_MODE,
+		SSD1306_CONTROL_BYTE_CMD,
+		SSD1306_SET_COLUMN_ADDRESS,
+		SSD1306_CONTROL_BYTE_CMD,
+		0,
+		SSD1306_CONTROL_BYTE_CMD,
+		(SSD1306_PANEL_NUMOF_COLUMS - 1),
+		SSD1306_CONTROL_BYTE_CMD,
+		SSD1306_SET_PAGE_ADDRESS,
+		SSD1306_CONTROL_BYTE_CMD,
+		0,
+		SSD1306_CONTROL_LAST_BYTE_CMD,
+		(SSD1306_PANEL_NUMOF_PAGES - 1)
+	};
+
+	if (i2c_write(driver->i2c, cmd_buf, sizeof(cmd_buf),
+		      CONFIG_SSD1306_I2C_ADDR)) {
+		LOG_ERR("Failed to write command");
+		return -1;
+	}
+
+	return i2c_burst_write(driver->i2c, CONFIG_SSD1306_I2C_ADDR,
+			       SSD1306_CONTROL_LAST_BYTE_DATA,
+			       (u8_t *)buf, desc->buf_size);
+
+#elif defined(CONFIG_SSD1306_SH1106_COMPATIBLE)
+	if (len != SSD1306_PANEL_NUMOF_PAGES * SSD1306_PANEL_WIDTH) {
+		return -1;
+	}
+
+	for (size_t pidx = 0; pidx < SSD1306_PANEL_NUMOF_PAGES; pidx++) {
+		if (ssd1306_write_page(dev, pidx, buf, SSD1306_PANEL_WIDTH)) {
+			return -1;
+		}
+		buf = (u8_t *)buf + SSD1306_PANEL_WIDTH;
+	}
+#endif
+
+	return 0;
+}
+
+static int ssd1306_read(const struct device *dev, const u16_t x,
+			const u16_t y,
+			const struct display_buffer_descriptor *desc,
+			void *buf)
+{
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+}
+
+static void *ssd1306_get_framebuffer(const struct device *dev)
+{
+	LOG_ERR("Unsupported");
+	return NULL;
+}
+
+static int ssd1306_set_brightness(const struct device *dev,
+				  const u8_t brightness)
+{
+	LOG_WRN("Unsupported");
+	return -ENOTSUP;
+}
+
+int ssd1306_set_contrast(const struct device *dev, const u8_t contrast)
+{
+	struct ssd1306_data *driver = dev->driver_data;
+	u8_t cmd_buf[] = {
+		SSD1306_CONTROL_BYTE_CMD,
+		SSD1306_SET_CONTRAST_CTRL,
+		SSD1306_CONTROL_LAST_BYTE_CMD,
+		contrast,
+	};
+
+	return i2c_write(driver->i2c, cmd_buf, sizeof(cmd_buf),
+			 CONFIG_SSD1306_I2C_ADDR);
+}
+
+static void ssd1306_get_capabilities(const struct device *dev,
+				     struct display_capabilities *caps)
+{
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = SSD1306_PANEL_WIDTH;
+	caps->y_resolution = SSD1306_PANEL_HEIGHT;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO10;
+	caps->current_pixel_format = PIXEL_FORMAT_MONO10;
+	caps->screen_info = SCREEN_INFO_MONO_VTILED;
+}
+
+static int ssd1306_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pf)
+{
+	LOG_ERR("Unsupported");
```

Comment:
Do not report/return an error in case supported pixel format (PIXEL_FORMAT_MONO10) is set